AdvancedDLSupport/SymbolTransformation/SymbolTransformer.cs

```diff
@@ -105,29 +105,64 @@ private string Transform
                 }
                 case Pascalize:
                 {
-                    return concatenated.Pascalize();
+                    return PascalizeValue(concatenated);
                 }
                 case Camelize:
                 {
-                    return concatenated.Camelize();
+                    return CamelizeValue(concatenated);
                 }
                 case Underscore:
                 {
-                    return concatenated.Underscore();
+                    return UnderscoreValue(concatenated);
                 }
                 case Dasherize:
                 {
-                    return concatenated.Dasherize();
+                    return DasherizeValue(concatenated);
                 }
                 case Kebaberize:
                 {
-                    return concatenated.Kebaberize();
+                    return KebaberizeValue(concatenated);
                 }
                 default:
                 {
                     throw new ArgumentOutOfRangeException(nameof(method), method, null);
                 }
             }
         }
+
+        private string CamelizeValue(string input)
+        {
+            var word = PascalizeValue(input);
+            return word.Length > 0 ? word.Substring(0, 1).ToLower() + word.Substring(1) : word;
+        }
+
+        private string PascalizeValue(string input)
+        {
+            return Regex.Replace(input, "(?:^|_)(.)", match => match.Groups[1].Value.ToUpper());
```

Comment:
Since these regexes don't change at runtime, they should be private static readonly fields, with the Compiled option specified.